src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -39,9 +39,50 @@ class Connection
      */
     private $amqpQueue;
 
-    public function __construct(array $connectionCredentials, array $exchangeConfiguration, array $queueConfiguration, bool $debug = false, AmqpFactory $amqpFactory = null)
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $amqpRetryExchange;
+
+    /**
+     * Available options:
```

Comment:
Yes, I think it's fine like this (but not ideal, true). Otherwise we'll need something as similar as the `SecurityFactoryInterface` in order to make it extendable to any transport with similar use-cases.